solution/index.js

```diff
@@ -0,0 +1,541 @@
+    const section = target.closest("section")
+    thisNewTaskButton = section.querySelector(".new-task-button")
+    document.addEventListener("keydown", enterToAddNewTask)
+}
+function enterToAddNewTask(event){
+    if (event.key === "Enter"){
+        event.preventDefault()
+        thisNewTaskButton.click()
+        thisNewTaskButton = null
+        document.removeEventListener("keydown", enterToAddNewTask)
+    }
+}
+
+//  Initializing DOM  //
+function generateLists(){
+    for (const task of taskObj.todo) {
+        const toDoList = document.querySelectorAll(".to-do-tasks")[0]
+        const newTask = createTask(task)
+        toDoList.appendChild(newTask)
+    }
+    for (const task of taskObj["in-progress"]) {
+        const inProgressList = document.querySelectorAll(".in-progress-tasks")[0]
+        const newTask = createTask(task)
+        inProgressList.appendChild(newTask)
+    }
+    for (const task of taskObj.done) {
+        const doneList = document.querySelectorAll(".done-tasks")[0]
+        const newTask = createTask(task)
+        doneList.appendChild(newTask)
+    }
+}
+generateLists()
+
+function generateLabelsForTasks(){
+    const tasksList = document.querySelectorAll(".task")
+    for (const task of tasksList) {
+        const taskUl = task.closest("ul")
+        if (taskUl.classList.contains("to-do-tasks")){
+            for (const labeledTasks in labeledTasksObj.todo){
+                if (task.innerText === labeledTasks){
+                    task.setAttribute("data-label", labeledTasksObj.todo[labeledTasks])
+                    task.setAttribute("style", `background-color:${labels[labeledTasksObj.todo[labeledTasks]]}`)
+                }
+            }
+        }
+        if (taskUl.classList.contains("in-progress-tasks")){
+            for (const labeledTasks in labeledTasksObj["in-progress"]){
+                if (task.innerText === labeledTasks){
+                    task.setAttribute("data-label", labeledTasksObj["in-progress"][labeledTasks])
+                    task.setAttribute("style", `background-color:${labels[labeledTasksObj["in-progress"][labeledTasks]]}`)
+                }
+            }
+        }
+        if (taskUl.classList.contains("done-tasks")){
+            for (const labeledTasks in labeledTasksObj.done){
+                if (task.innerText === labeledTasks){
+                    task.setAttribute("data-label", labeledTasksObj.done[labeledTasks])
+                    task.setAttribute("style", `background-color:${labels[labeledTasksObj.done[labeledTasks]]}`)
+                }
+            }
+        }
+    }
+}
+generateLabelsForTasks()
+// Clear tasks
+function clearTasks(){
+    const tasksElems = document.querySelectorAll(".task")
+    for (const taskElem of tasksElems){
+        taskElem.parentElement.removeChild(taskElem)
+        removeTaskData(taskElem.innerText)
+        updateLocalSaveData()
+    }
+}
+
+//  Maniputale tasks order and sections  //
+let selectedTask
```

Comment:
Global variables create messy code. You can find another way to know which element is hovered.